src/Components/Common/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,850 @@
+    setMarkdown(newValue);
+
+    const newLineStart =
+      lines.slice(0, lineIndex).join("\n").length + (lineIndex > 0 ? 1 : 0);
+    const newCursorPosition = newLineStart + newText.length;
+    editorRef.current.setSelectionRange(newCursorPosition, newCursorPosition);
+    editorRef.current.focus();
+  };
+
+  const getCurrentLineIndex = (cursorPosition: number) => {
+    const text = editorRef.current?.value || "";
+    return text.substring(0, cursorPosition).split("\n").length - 1;
+  };
+
+  const formatUrl = (url: string) => {
+    if (!/^https?:\/\//i.test(url)) {
+      return `https://${url}`;
+    }
+    return url;
+  };
+  const handleLink = () => {
+    if (!editorRef.current) return;
+
+    const start = editorRef.current.selectionStart;
+    const end = editorRef.current.selectionEnd;
+    const text = editorRef.current.value;
+
+    const selectedText = text.substring(start, end);
+
+    setLinkDialogState({
+      showDialog: true,
+      url: "",
+      linkText: selectedText,
+      selectedText,
+    });
+  };
+
+  const handleInsertLink = () => {
+    if (!editorRef.current || !linkDialogState.url.trim()) return;
+
+    const { start } = getCaretCoordinates(
+      editorRef.current,
+      editorRef.current.selectionStart,
+    );
+
+    const text = editorRef.current.value;
+
+    const beforeSelection = text.substring(0, start);
+    const afterSelection = text.substring(
+      start + linkDialogState.selectedText.length,
+    );
+
+    const markdownLink = `[${linkDialogState.linkText || linkDialogState.url}](${formatUrl(linkDialogState.url)})`;
+    const newText = `${beforeSelection}${markdownLink}${afterSelection}`;
+
+    setMarkdown(newText);
+    editorRef.current.focus();
+    editorRef.current.setSelectionRange(
+      start + markdownLink.length,
+      start + markdownLink.length,
+    );
+
+    setLinkDialogState({
+      showDialog: false,
+      url: "",
+      linkText: "",
+      selectedText: "",
+    });
+  };
+
+  const uploadfile = async (data: CreateFileResponse, file: File) => {
+    const url = data.signed_url;
+    const internal_name = data.internal_name;
+    const newFile = new File([file], `${internal_name}`);
+    return new Promise<void>((resolve, reject) => {
+      uploadFile(
+        url,
+        newFile,
+        "PUT",
+        { "Content-Type": file.type },
+        async (xhr: XMLHttpRequest) => {
+          if (xhr.status >= 200 && xhr.status < 300) {
+            Notification.Success({
+              msg: "File Uploaded Successfully",
+            });
+            resolve();
+          } else {
+            Notification.Error({
+              msg: "Error Uploading File: " + xhr.statusText,
+            });
+            reject();
+          }
+        },
+        null,
+        () => {
+          Notification.Error({
+            msg: "Error Uploading File: Network Error",
+          });
+          reject();
+        },
+      );
+    });
+  };
```

Comment:
we already have the `uploadFile` method as a shared utility function. Why duplicate it here again?

Reply:
just upadated to use existing hook (i am not aware of it since it got recently updated)